src/System.Xml.XmlDocument/src/System/Xml/Dom/XmlElement.cs

```diff
@@ -58,8 +58,14 @@ public override XmlNode CloneNode(bool deep)
                 foreach (XmlAttribute attr in Attributes)
                 {
                     XmlAttribute newAttr = (XmlAttribute)(attr.CloneNode(true));
-                    if (attr is XmlUnspecifiedAttribute && attr.Specified == false)
-                        ((XmlUnspecifiedAttribute)newAttr).SetSpecified(false);
+                    if (attr.Specified == false)
```

Comment:
@Maxwe11, thanks, is it possible for a 3rd-party to have created their own derived implementation that could do something else, or is external derivation prevented (e.g. via internal constructors)?